public/scripts/services/imageService.js

```diff
@@ -86,12 +86,12 @@ app.service('imageService', ['$http','$q', function ($http, $q) {
     }
 
     //Get All Images
-    this.getAllImages = function(start, end){
-        return $http.get("/api/images?itemsPerPage=20&sortBy=uploadedOn");
+    this.getAllImages = function(){
+        return $http.get("/api/images/");
     }
     //Pagination and Sorting
-    this.getImagesPerPage = function(start, pageNumber){
-        return $http.get("/api/images?itemsPerPage="+start+"&pageNumber=1&sortBy=uploadedOn");
+    this.getImagesPerPage = function(itemsPerPage, pageNumber){
```

Comment:
Agreed, but isn't this service the same service that should be used?

Consider this approach based on Rosario's suggestion:

```javascript
this.getImages = function(options) {
  var url = '/api/images?';
  if (options) {
    for (var property in options) {
      url += property + "=" + options[property] + "&";
    }
  }
  return $http.get(url);
}

// you can call it like this:
service.getImages(); // all images
service.getImages({ itemsPerPage: 10, pageNumber: 1 }); // first page of 10 images
service.getImages({ sortBy: "uploadedOn" }); // all images, sorted
service.getImages({ itemsPerPage: 10, pageNumber 1:, sortBy: "uploadedOn" }); // first page, 10 images, sorted
```